packages/docregistry/src/registry.ts

```diff
@@ -1508,14 +1508,23 @@ export interface IDocumentWidget<
   readonly content: T;
 
   /**
-   * A promise resolving after the content widget is revealed.
+   * The context associated with the document.
    */
-  readonly revealed: Promise<void>;
+  readonly context: DocumentRegistry.IContext<U>;
 
   /**
-   * The context associated with the document.
+   * Whether the document has an auto-generated name or not.
+   *
+   * #### Notes
+   * A document has auto-generated name if its name is untitled and up
+   * to the instant the user saves it manually for the first time.
    */
-  readonly context: DocumentRegistry.IContext<U>;
+  hasAutoName?: boolean;
```

Comment:
This attribute is not clear to me. Does this have any relation to `renameUntitledFileOnSave`? Is there a random auto-generated name suggested if this option is selected?

Reply:
I started with `isUntitled` then switch hoping to support the document copy feature too. But that one is more tricky. So let's go back to my initial proposition.